torch/testing/_internal/common_methods_invocations.py

```diff
@@ -5672,6 +5672,29 @@ def sample_inputs_pixel_unshuffle(op_info, device, dtype, requires_grad, **kwarg
         for downscale_factor in (1, 3)
     ]
 
+def sample_inputs_kl_div(op_info, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    shapes_and_kwargs: List[Tuple[Tuple[int, ...], Dict[str, Any]]] = [
+        ((2,), dict()),
+        ((2, 3), dict()),
+        ((2, 3, 4), dict()),
+        ((2,), dict(log_target=True)),
+        ((2,), dict(reduction="none")),
+        ((2,), dict(reduction="batchmean")),
+        ((2,), dict(reduction="sum")),
```

Comment:
Given that the `log_target` takes different routes in the backward, might be good to test the 8 combination log_target x reduction?